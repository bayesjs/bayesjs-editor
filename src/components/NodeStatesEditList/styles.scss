.node-states-edit-list {
  list-style: none;
  margin: 0;
  padding: 2px 5px;
  height: 200px;
  border: solid 1px #333;
  overflow: auto;
}
