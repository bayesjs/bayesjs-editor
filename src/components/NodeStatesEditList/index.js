import React from 'react';
import NodeStateEdit from 'components/NodeStateEdit';
import PropTypes from 'prop-types';
import { statePropTypes } from 'models';
import { bem } from 'utils/styles';
import styles from './styles.scss';

const componentClassName = bem(styles);

const NodeStatesEditList = ({ states, onDeleteState }) => (
  <ul className={componentClassName}>
    {states.map((state, index) => (
      <NodeStateEdit
        key={state}
        state={state}
        index={index}
        onDeleteState={onDeleteState}
      />
    ))}
  </ul>
);

NodeStatesEditList.propTypes = {
  states: PropTypes.arrayOf(statePropTypes).isRequired,
  onDeleteState: PropTypes.func.isRequired,
};

export default NodeStatesEditList;
