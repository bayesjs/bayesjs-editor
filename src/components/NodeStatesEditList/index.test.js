import React from 'react';
import { render } from '@testing-library/react';
import NodeStatesEditList from './index';

const renderComponent = (props) => {
  const { container } = render(
    <NodeStatesEditList {...props} />,
  );

  return container.firstChild;
};

describe('NodeStatesEditList Component', () => {
  const defaultProps = {
    states: [
      'State 1',
      'State 2',
      'State 3',
    ],
    onDeleteState: jest.fn(),
  };
  let component;

  beforeEach(() => {
    component = renderComponent(defaultProps);
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
