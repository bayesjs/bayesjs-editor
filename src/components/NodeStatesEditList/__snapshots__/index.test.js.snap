// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeStatesEditList Component matches snapshot 1`] = `
<ul
  class=""
>
  <li
    class=""
    data-testid="node-state-edit__state-1"
  >
    <span
      title="State 1"
    >
      State 1
    </span>
    <button
      class="__button"
      data-testid="button__remove-state"
      title="Remover Estado"
      type="button"
    >
      <i
        class="fa fa-trash"
      />
    </button>
  </li>
  <li
    class=""
    data-testid="node-state-edit__state-2"
  >
    <span
      title="State 2"
    >
      State 2
    </span>
    <button
      class="__button"
      data-testid="button__remove-state"
      title="Remover Estado"
      type="button"
    >
      <i
        class="fa fa-trash"
      />
    </button>
  </li>
  <li
    class=""
    data-testid="node-state-edit__state-3"
  >
    <span
      title="State 3"
    >
      State 3
    </span>
    <button
      class="__button"
      data-testid="button__remove-state"
      title="Remover Estado"
      type="button"
    >
      <i
        class="fa fa-trash"
      />
    </button>
  </li>
</ul>
`;
